### PR TITLE
New components and hooks added

### PR DESCRIPTION
**Pull request by jeezOC**

### CODE REVIEW

File: src/components/EventBanner/EventBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import useEvents from '../../hooks/useEvents';
+import { Box, SimpleGrid, GridItem, Stack, Text, Button, Flex, Heading, Spacer, useBoolean } from "@chakra-ui/react";
+
+const EventBanner = () => {
+  const eventsContex = useEvents();
```
Comment (hermosasoftware) [inline src/components/EventBanner/EventBanner.tsx:6]: eventsContext. Make sure you add the "t" at the end in all the missing places

File: src/components/EventBanner/EventBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const { currentEvent } = eventsContex;
+
+  const getStartDate = () => {
```
Comment (hermosasoftware) [inline src/components/EventBanner/EventBanner.tsx:9]: Replace this with a dayjs method

File: src/components/EventCard.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+interface props {
+    event?: any;
+    index?: any;
+}
+const EventCard = ({ event, index }: props) => {
+    const navigate = useNavigate();
+
+    const eventsContex = useEvents();
+    const handleOnClick = () => {
+        // eventsContex.setCurrentEvent(event);
```
Comment (hermosasoftware) [inline src/components/EventCard.tsx:14]: remove comment

File: src/components/EventCard.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        eventsContex.setCurrentEvent(event);
+        navigate('event-details');
+    }
+    const getStartDate = () => {
```
Comment (hermosasoftware) [inline src/components/EventCard.tsx:18]: Replace this with a dayjs method

File: src/components/EventCard.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        event &&
+        <Box maxWidth='400px' minWidth='90%' borderRadius='10' borderWidth='3px' borderColor={'teal.200'} p='4' onClick={handleOnClick}>
+            <Box maxWidth='100%' minWidth='50%' minHeight={'200px'} borderRadius='10' borderWidth='3px' borderColor={'teal.200'} >
+            </Box>
+            <Stack spacing={7}>
+                <Heading size='lg' as='h2' >
+                    {event?.name}
+                </Heading >
+
+                {/* <Text as='sub' fontSize='lg' >{event?.location} </Text> */}
```
Comment (hermosasoftware) [inline src/components/EventCard.tsx:44]: remoive comment

File: src/components/EventCard.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    const getEndDate = () => {
+        const date = new Date(event?.dates[1].toDate());
+        return (date.getDate() +
+            "/" + (date.getMonth() + 1) +
+            "/" + date.getFullYear() +
+            " " + date.getHours() +
+            ":" + (date.getMinutes() < 10 ? '0' + date.getMinutes() : date.getMinutes()));
+    }
+    return (
+        event &&
```
Comment (hermosasoftware) [inline src/components/EventCard.tsx:35]: remove this validation

File: src/components/EventCard.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import useEvents from "../hooks/useEvents";
+import { useNavigate } from "react-router-dom";
+
+interface props {
+    event?: any;
+    index?: any;
+}
+const EventCard = ({ event, index }: props) => {
+    const navigate = useNavigate();
+
```
Comment (hermosasoftware) [inline src/components/EventCard.tsx:11]: unnecessary empty line

File: src/components/EventCard.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+                    <Spacer />
+                    <Stack direction='row' spacing={0} flexWrap='nowrap'>
+                        <Text as='sub' fontSize='lg' fontWeight={'bold'} wordBreak={'keep-all'}>
+                            {'End:'} </Text>
+                        <Text as='sub' fontSize='lg' color='gray.400' wordBreak={'keep-all'} >
+                            {getEndDate().toString()} </Text>
+                    </Stack>
+                </Stack>
+            </Stack>
+
```
Comment (hermosasoftware) [inline src/components/EventCard.tsx:61]: empty lines

File: src/components/FormInput/FormInput.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    name?: string;
+    value:any;
+    onChange:any;
+    isInvalid?:boolean;
+    errors?:any;
+}
+
+const FormInput = ({formLabel, type, helperText, size, isRequired, name, value, onChange, isInvalid,errors}:props) => {
+    return (
+        <>
```
Comment (hermosasoftware) [inline src/components/FormInput/FormInput.tsx:19]: Fix all file indentation

File: src/components/Navigation.tsx [hunk 10]
```diff
@@ -1,20 +1,64 @@
...
-  </Flex>
-);
+const Navigation = () => {
+  const navigate = useNavigate();
+  const navigateTo = (path: any) => {
+    navigate(path);
+  }
+  return (
+    <Flex minWidth='max-content' alignItems='center' gap='2' justifyContent="center" padding="6" >
+      {/* <Menu>
```
Comment (hermosasoftware) [inline src/components/Navigation.tsx:36]: Remove commented out code

File: src/context/EventsContext.tsx [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import { createContext, useState } from 'react'
```
Comment (hermosasoftware) [inline src/context/EventsContext.tsx:1]: fix all file indentation

File: src/services/eventsService.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { collection, getDocs } from '@firebase/firestore'
+import {db} from './firebaseService'
```
Comment (hermosasoftware) [inline src/services/eventsService.ts:2]: space after and before brakets like in line 1

File: src/services/eventsService.ts [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+
+const eventsCollectionRef = collection(db, 'events')
+
+export const getEvents = async()=>{
```
Comment (hermosasoftware) [inline src/services/eventsService.ts:6]: please add spaces as such: async () => {

File: src/services/eventsService.ts [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+
+    const data = await getDocs(eventsCollectionRef)
+    return (data.docs.map((doc)=>({...doc.data(), id:doc.id})));
```
Comment (hermosasoftware) [inline src/services/eventsService.ts:9]: space after `id : doc`

File: src/services/firebaseService.ts [hunk 15]
```diff
@@ -1,21 +1,25 @@
 // Import the functions you need from the SDKs you need
```
Comment (hermosasoftware) [inline src/services/firebaseService.ts:1]: fix all file indentation

File: src/components/EventCard.tsx [hunk 16]
```diff
@@ -1,68 +1,44 @@
...
-
-        </Box>
-    )
+  const handleOnClick = () => {
+    eventsContex.setCurrentEvent(event);
+    navigate('generate-code');
+  }
+  return (
+    <Box as="button" maxWidth='500px' minWidth='450px' borderRadius='10' borderWidth='3px' borderColor={'teal.200'} p='4' onClick={handleOnClick} _hover={{ bg: "blue.900" }}>
+      <Box maxWidth='100%' minWidth='50%' minHeight={'200px'} borderRadius='10' borderWidth='3px' borderColor={'teal.200'} >
```
Comment (Chinaski1979) [inline src/components/EventCard.tsx:19]: I'm not sure I understand this component, is not wrapping any other element.

File: src/services/eventsService.ts [hunk 17]
```diff
@@ -1,10 +1,9 @@
...
+const eventsCollectionRef = collection(db, "events");
 
-export const getEvents = async()=>{
-
-    const data = await getDocs(eventsCollectionRef)
-    return (data.docs.map((doc)=>({...doc.data(), id:doc.id})));
-}
\ No newline at end of file
+export const getEvents = async () => {
+	const data = await getDocs(eventsCollectionRef);
```
Comment (Chinaski1979) [inline src/services/eventsService.ts:7]: still incorrect indentation

File: src/components/EventCard.tsx (hunk 16 above)
Comment (jeezOC) [reply src/components/EventCard.tsx:19]: No is not, is blank space to maybe display an image
should I delete it for now?

File: src/components/TicketSelection.tsx [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+  const { aquiredTickets } = eventsContext;
+
+  const handleOnChangeRB = (e: any) => {
+    console.log(e.valueOf())
+    setRadioValue(e.valueOf())
+  }
+  const handleAdd = (value: any) => {
+    let tempTicket: any = {};
+    let tempTicketList = [];
+    const tktExists = eventsContext.aquiredTickets?.find((ticket: any) => ticket.name === value) || false;
```
Comment (Chinaski1979) [inline src/components/TicketSelection.tsx:22]: change find for findIndex.

```
const tktIndex = eventsContext.aquiredTickets?.findIndex((ticket: any) => ticket.name === value

const aquiredTicketsCopy = eventsContext.aquiredTickets.slice();
// only if tktIndex >= 0
aquiredTicketsCopy[tktIndex].quantity ++; 
else {
  aquiredTicketsCopy.push(selectedTicket);



const months = ['Jan', 'March', 'April', 'June'];
months.splice(1, 0, 'Feb');
// inserts at index 1
console.log(months);
// expected output: Array ["Jan", "Feb", "March", "April", "June"]

```

File: src/components/TicketSelection.tsx [hunk 19]
```diff
@@ -16,33 +16,30 @@ const TicketSelection = () => {
...
-      tempTicketList = [...tempTicketList, tempTicket];
+    });
+    return tempTicket;
+  }
+
+  const handleAdd = (ticketName: string) => {
+    const tktIndex = eventsContext.aquiredTickets?.findIndex((ticket: any) => ticket.name === ticketName);
+    let aquiredTicketsCopy = eventsContext.aquiredTickets.slice();
+    if (tktIndex>=0) {
+      aquiredTicketsCopy[tktIndex].cant++;
```
Comment (Chinaski1979) [inline src/components/TicketSelection.tsx:35]: remember to change "cant" to "quantity"

**Review by Chinaski1979 — APPROVED** [review]

**Pull request merged**